compiler+runtime/include/cpp/clojure/core_native.hpp

```diff
@@ -1,5 +1,21 @@
 #pragma once
 
 #include <jank/c_api.h>
+#include <jank/runtime/object.hpp>
 
 extern "C" jank_object_ref jank_load_clojure_core_native();
+
+namespace clojure::core_native
```

Comment:
(I know this is a draft, so I'm not doing a full review. Just a note)

We ultimately want to get rid of `core_native` entirely. This stuff should go under `runtime/core.hpp` until it fits under another grouping like `runtime/core/seq.hpp`.

Reply:
Just for the new things. We don't want to be adding anything new to `core_native`.

Each of the `foo_native` files was a temporary solution, before jank had good interop. They should all be going away, with their functions just being a part of jank's runtime.

Reply:
My current plan is to work towards removing all of the `clojure.core-native` vars in clojure.core, then refactor the cpp code to be better organized 👍

Reply:
Are you okay with these functions living in this namespace until `clojure.core-native` is removed? I haven't added any new functions, simply made current ones non-static to ease interop migration and make it easier to follow / verify.

Reply:
Yeah, that's ok.